### PR TITLE
(doc) Update index.

### PR DESCRIPTION
**Pull request by gguillotte**
- Add static file content API endpoint doc.
- Add new auth and logging docs.
- Convert headings to sentence case per recent changes to company style.
- Structure introduction and usage docs into sections.
- Add note pointing to PE's docs.

### CODE REVIEW

**Comment by gguillotte** [comment]
CC @nfagerlund as most of these changes would need porting over to all the places where Server's sidebar navs live on the Docs site if accepted.

**Comment by KevinCorcoran** [comment]
I'd appreciate it if we could let #1049 land before merging this, since it contains many docs changes, in the interest of avoiding any more conflicts in that already-huge merge-up.

**Comment by gguillotte** [comment]
@KevinCorcoran Totally agreed. This is a very low priority.

File: documentation/index.markdown [hunk 1]
```diff
@@ -6,37 +6,43 @@ canonical: "/puppetserver/latest/"
...
+    * [External SSL termination](./external_ssl_termination.markdown)
+    * [Tuning guide](./tuning_guide.markdown)
+    * [Restarting Puppet Server](./restarting.markdown)
+* **Known issues and workarounds**
+    * [Known issues](./known_issues.markdown)
+    * [SSL problems with load-balanced PuppetDB servers ("Server Certificate Change" error)](./ssl_server_certificate_change_and_virtual_ips.markdown)
+* **Administrative API endpoints**
+    * [Environment cache](./admin-api/v1/environment-cache.markdown)
+    * [JRuby pool](./admin-api/v1/jruby-pool.markdown)
+* **Server-specific Puppet API endpoints**
```
Comment (KevinCorcoran) [inline documentation/index.markdown]: What do you think about combining this section with the one immediately above?

**Comment by KevinCorcoran** [comment]
👍  looks fine to me, just had the one question

File: documentation/index.markdown (hunk 1 above)
Comment (gguillotte) [inline documentation/index.markdown]: @KevinCorcoran I can do that, though I'd still like to indicate that the admin and Puppet endpoints are at different locations. I'll figure out a way to do this.

**Comment by rlinehan** [comment]
@gguillotte what's the status on this? looks like @KevinCorcoran had a question a while ago, wasn't sure if you were going to update this or if you would prefer we merge it as is. I'm 👍 , fwiw.

**Comment by gguillotte** [comment]
@rlinehan I think we can merge this as-is, and I can follow on once I figure out structural changes to the API docs.

**Pull request merged**